### DISCUSSION TITLE
Addition of Iron pieces

### DISCUSSION THREAD

**Discussion by ArsenalAlex108** [fairy-stockfish/Fairy-Stockfish#604] [Ideas]
This feature can partially be used by setting extinctionValue = win, though capturing the supposed "iron" piece is still a legal:

[ExampleBoard]
king = k
knight = n
bishop = b
rook = r
queen = q

pawn = p

doubleStep = false

customPiece2 = s:mK

maxFile = 5
maxRank = 6

promotionPieceTypes = q
promotedPieceType = n:q b:q r:q
promotionRank = 6
mandatoryPiecePromotion = true

extinctionValue = win
extinctionPieceTypes = s

startFen = qkbnr/ppspp/5/5/PPSPP/QKBNR

I was wondering if it is possible to separate iron pieces into a feature separated from extinction pieces, so that both iron pieces and extinction pieces can exist on the same board?

**Comment by ianfab** [comment]
By "iron" piece I assume you are referring to pieces that are uncapturable, i.e., that capturing them is illegal, but other than that act like any other piece, right?

**Reply by ArsenalAlex108** [reply]
Yes of course, that's all there is to it!

**Reply by ianfab** [reply]
I converted it to an issue to track it as a feature request.
